src/main/scala/ai/privado/languageEngine/java/threatEngine/PIIShouldNotBePresentInMultipleTables.scala

```diff
@@ -0,0 +1,79 @@
+package ai.privado.languageEngine.java.threatEngine
+
+import ai.privado.cache.TaggerCache
+import ai.privado.exporter.ExporterUtility
+import ai.privado.languageEngine.java.passes.read.EntityMapper
+import ai.privado.languageEngine.java.threatEngine.ThreatUtility.{getSourceNode, hasDataElements}
+import ai.privado.model.PolicyOrThreat
+import ai.privado.model.exporter.{DataFlowSubCategoryPathExcerptModel, ViolationProcessingModel}
+import io.shiftleft.codepropertygraph.generated.Cpg
+import io.shiftleft.semanticcpg.language._
+import org.slf4j.LoggerFactory
+
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+import scala.util.Try
+
+object PIIShouldNotBePresentInMultipleTables {
+
+  private val logger = LoggerFactory.getLogger(getClass)
+
+  /** Check for violation for data leakage to logs threat - consumes already generated dataflows
+    *
+    * @param cpg
+    *   cpg
+    * @return
+    */
+  def getViolations(
+    threat: PolicyOrThreat,
+    cpg: Cpg,
+    taggerCache: TaggerCache
+  ): Try[(Boolean, List[ViolationProcessingModel])] = Try {
+    if (hasDataElements(cpg)) {
+      val violatingFlows        = ListBuffer[ViolationProcessingModel]()
+      val entityMapper          = EntityMapper.getClassTableMapping(cpg)
+      val dataElementToTableMap = mutable.HashMap[String, ListBuffer[String]]()
+
+      entityMapper.foreach((entityWithTable) => {
+        val tableName        = entityWithTable._1
```

Comment:
I see this code getting repeated in multiple threat, create a common functions where ever possible

Reply:
Let me check, I think code inside is different to solve different usecases